libs/bounty/src/bounty.service.ts

```diff
@@ -29,4 +29,25 @@ export class BountyService extends TypeOrmCrudService<Bounty> {
       times: Not('0'),
     });
   }
+
+  async getLastBountyClaim(
+    bountyId: string,
+    accountId: string,
+    untilTimestamp?: number,
+  ): Promise<BountyClaim | null> {
+    const bounty = await this.bountyRepository.findOne(bountyId, {
+      relations: ['bountyClaims'],
+    });
+    return bounty.bountyClaims.reduce((lastClaim, bountyClaim) => {
+      if (
+        bountyClaim.accountId === accountId &&
+        (!untilTimestamp || Number(bountyClaim.startTime) < untilTimestamp) &&
```

Comment:
can we extract this statement into meaningful function?

Reply:
please check updated function